### ISSUE TITLE
Test failed: sequence item 1: expected str instance, bytes found

### ISSUE THREAD

**Issue by sbjeffrey** [s3tools/s3cmd#1349]
I'm trying to use the s3tools to connect to the Digital Ocean's Spaces, but I keep getting this error and can't figure out what the root cause is.
Im using s3cmd 2.3.0 and python 3.9.6

```
Please wait, attempting to list all buckets...
ERROR: Test failed: sequence item 1: expected str instance, bytes found
```

I've tried to debug the error using `s3cmd ls --debug`, but still can't figure out the exact reason.

I've got the following settings, based on the [documentation](https://docs.digitalocean.com/products/spaces/reference/s3cmd/).

``` 
  Access Key: <hidden>
  Secret Key: <hidden>
  Default Region: AMS3  // (I also tried US and AMS)
  S3 Endpoint: ams3.digitaloceanspaces.com
  DNS-style bucket+hostname:port template for accessing a bucket: %(bucket)s.ams3.digitaloceanspaces.com
  Encryption password: <hidden>
  Path to GPG program: /opt/homebrew/bin/gpg
  Use HTTPS protocol: True
  HTTP Proxy server name:
  HTTP Proxy server port: 0
```

Does anyone know what the solution is for this error?

**Comment by jbilandzija** [comment]
Had the same output like @erikamag having s3cmd installed via Homebrew. I removed it & reinstalled via pip (`pip3 install s3cmd`) - now it runs fine. Hopefully it resolves your issue as well.

**Comment by sbjeffrey** [comment]
> Had the same output like @erikamag having s3cmd installed via Homebrew. I removed it & reinstalled via pip (`pip3 install s3cmd`) - now it runs fine. Hopefully it resolves your issue as well.

Thanks! I've got it working now!
(Had to add the s3cmd executable to my systems PATH first.)

**Comment by pozymasika** [comment]
> (Had to add the s3cmd executable to my systems PATH first.)

In case anyone bumps into this, here's some info on adding the executable to your path on Mac

First get your python version
```
python3 --version
Python 3.9.6
```
then opening your `.bashrc` or `.zshrc`
```
nano ~/.zshrc
```

and adding this (replace `<username>/` with your username)
```
export PATH="/Users/<username>/Library/Python/3.9/bin:$PATH"
```

then after saving the file, finally running

```
source ~/.zshrc
```

**Comment by rampageservices** [comment]
This PR fixes this issue for Python 3.12 installed via brew.
https://github.com/s3tools/s3cmd/pull/1355
